File: 118. Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int nr) {
        vector<vector<int>> arr;
        vector<int> temp;
        temp.push_back(1);
        arr.push_back(temp);

        if (nr == 1) return arr;
        temp.push_back(1);
        arr.push_back(temp);

        if (nr == 2) return arr;

        for (int i = 2; i < nr; i++) {
            vector<int> level;
            level.push_back(1);
            int left = 0, right = 1;
            int size = arr[i - 1].size();
            while (right < size) {
                level.push_back(arr[i - 1][left] + arr[i - 1][right]);
                left++;
                right++;
            }
            level.push_back(1);
            arr.push_back(level);
        }

        return arr;
    }
};

File: 118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        for(int i = 0; i < numRows; i++){
            List<Integer> temp = new ArrayList<>();
            temp.add(1);
            if(i == 0){
                ans.add(temp);
                continue;
            }
            if(i == 1){
                temp.add(1);
                ans.add(temp);
                continue;
            }
            
            for(int j = 1; j < i; j++) temp.add(ans.get(i - 1).get(j) + ans.get(i - 1).get(j - 1));
            temp.add(1);
            ans.add(temp);
        }
        return ans;
    }
}